bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/Proposal.scala

```diff
@@ -50,6 +53,14 @@ object Proposal {
 
   }
 
+  // find the target of opportunity of the proposal if possible
+  def toO(p: Option[Proposal]): Option[TooOption] = p.map(proposalClass.get) match {
+    case Some(q: QueueProposalClass)         => q.tooOption.some
+    case Some(l: LargeProgramClass)          => l.tooOption.some
+    case Some(f: FastTurnaroundProgramClass) => f.tooOption.some
+    case _                                   => none
+  }
```

Comment:
This isn't your doing nor do I think you should change anything, but I'll comment that I wish we (probably it was me) had chosen a different name for `TooOption`.  It gets confusing when mixed with Scala `Option`.